### PR TITLE
MM-18679 : added test for post options menu in mobile view

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary

- Added the e2e test case for Mobile view to test if post menu and its options are available.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12268
Jira https://mattermost.atlassian.net/browse/MM-18679

### CODE REVIEW

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Mobile view Post menu options button in RHS', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]: Please description as we're using this for test reporting.
```suggestion
describe('Messaging', () => {
```

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        cy.getLastPostId().then((postId) => {
+            cy.clickPostCommentIcon(postId);
+        });
+
+        // # Enter valid text into RHS
+        const replyValid = 'Hi ' + Date.now();
+        cy.postMessageReplyInRHS(replyValid);
+    });
+
+    it('should always be visible in the RHS for last posts and should open options modal.', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]:
```suggestion
    it('M18679 - Mobile view: Post options menu (3-dots) is present on a reply post in RHS', () => {
```

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        // # Get the last entered RHS post
+        cy.getLastPostId().then((lastPostId) => {
+            const dotMenuButtonID = `#RHS_COMMENT_button_${lastPostId}`;
+
+            // * Check to see if button is visible
+            cy.get(dotMenuButtonID).should('be.visible');
+
+            // # Click on the options menu
+            cy.get(dotMenuButtonID).click();
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]: Above can be combined into
```suggestion
            cy.get(dotMenuButtonID).should('be.visible').click();
```

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+            const dropDownMenuOfPostOptionsID = `#RHS_COMMENT_dropdown_${lastPostId}`;
+
+            // * Check if modal for post options have opened
+            cy.get(dropDownMenuOfPostOptionsID).should('be.visible');
+
+            cy.get(dropDownMenuOfPostOptionsID).within(() => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]: Same here
```suggestion
            cy.get(dropDownMenuOfPostOptionsID).should('be.visible').within(() => {
```

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                // * Check if atleast 1 item is present
+                cy.get('li').should('have.length.greaterThan', 0);
+
+                // * Check if one of the options are as follows
+                cy.get('li').should('contain.text', 'Add Reaction');
+                cy.get('li').should('contain.text', 'Mark as Unread');
+                cy.get('li').should('contain.text', 'Permalink');
+                cy.get('li').should('contain.text', 'Flag');
+                cy.get('li').should('contain.text', 'Edit');
+                cy.get('li').should('contain.text', 'Delete');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]: I'd suggest to check those as if an actual user is looking. That's independent on structure.  Such could be:
```javascript
cy.findByText('Add Reaction').should('be.visible');
cy.findByText('Mark as Unread').should('be.visible');
cy.findByText('Permalink').should('be.visible');
cy.findByText('Flag').should('be.visible');
cy.findByText('Pin to channel').should('be.visible');
cy.findByText('Edit').should('be.visible');
cy.findByText('Delete').should('be.visible');
```

File: e2e/cypress/integration/messaging/post_options_menu_spec.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                cy.findByText('Mark as Unread').should('be.visible');
+                cy.findByText('Permalink').should('be.visible');
+                cy.findByText('Flag').should('be.visible');
+                cy.findByText('Pin to channel').should('be.visible');
+                cy.findByText('Edit').should('be.visible');
+                cy.findByText('Delete').should('be.visible');
+            });
+        });
+    });
+});
```
Comment (marianunez) [inline e2e/cypress/integration/messaging/post_options_menu_spec.js]: Nit: Missing new line

Comment (M-ZubairAhmed) [reply e2e/cypress/integration/messaging/post_options_menu_spec.js]: added :+1:

**Pull request merged**